audio/descriptors.c

```diff
@@ -42,81 +47,235 @@ __code __at(DSCR_AREA) struct usb_descriptors code_descriptors = {
         .bLength            = USB_DT_DEVICE_QUALIFIER_SIZE,
         .bDescriptorType    = USB_DT_DEVICE_QUALIFIER,
         .bcdUSB             = USB_BCD_V20,
-        .bDeviceClass       = 0, // Class and protocol defined per interface 
+        /* Class defined per interface */
+        .bDeviceClass       = 0,
         .bDeviceSubClass    = 0,
+        /* Protocol defined per interface */
         .bDeviceProtocol    = 0,
+        /* packet size is in kB */
         .bMaxPacketSize0    = 64,
         .bNumConfigurations = 1,
-        .bRESERVED          = 0, // Must be zero
+        /* Must be zero */
+        .bRESERVED          = 0,
     },
     .highspeed = {
         .config = {
             .bLength                = USB_DT_CONFIG_SIZE,
             .bDescriptorType        = USB_DT_CONFIG,
             .wTotalLength           = sizeof(descriptors.highspeed),
-            .bNumInterfaces         = 1,
+            .bNumInterfaces         = 2,
             .bConfigurationValue    = 1,
             .iConfiguration         = 0,
             .bmAttributes           = USB_CONFIG_ATT_ONE,
-            .bMaxPower              = 0x32, // * 2 mA
+            /* bMaxPower has a resolution of 2mA */
```

Comment:
Is this 0x32 == 2 * 0x32 == 50 * 2 == 100mA?

Probably shouldn't use hex here.... ```c
.bMaxPower = 100 / 2, /* bMaxPower is in 2mA increments */
```
Seems clearer?

Reply:
Fixed